src/libcore/atomic.rs

```diff
@@ -1089,3 +1090,21 @@ pub fn fence(order: Ordering) {
         }
     }
 }
+
+macro_rules! impl_Debug {
+    ($($t:ident)*) => ($(
+        impl fmt::Debug for $t {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                write!(f, "Atomic {{ {:?} }}", self.load(Ordering::Relaxed))
```

Comment:
I believe this should be `write!(f, concat!(stringify!($t), "{{ {:?} }}", ...)` so as to actually print which type this is.